src/main/java/com/devcycle/sdk/server/local/api/DVCLocalEventsApiClient.java

```diff
@@ -23,10 +25,25 @@ public final class DVCLocalEventsApiClient {
 
     private String eventsApiUrl;
 
-    private DVCLocalEventsApiClient(DVCLocalOptions options) {
+    public DVCLocalEventsApiClient(String sdkKey, DVCLocalOptions options) {
         OBJECT_MAPPER.setSerializationInclusion(JsonInclude.Include.NON_NULL);
         okBuilder = new OkHttpClient.Builder();
 
+        IRestOptions restOptions = options.getRestOptions();
+        if(restOptions != null)
+        {
+            if(restOptions.getHostnameVerifier() != null){
+                okBuilder.hostnameVerifier(restOptions.getHostnameVerifier());
+            }
+
+            if(restOptions.getSocketFactory() != null && restOptions.getTrustManager() != null){
+                okBuilder.sslSocketFactory(restOptions.getSocketFactory(), restOptions.getTrustManager());
+            }
+            okBuilder.addInterceptor(new CustomHeaderInterceptor(restOptions));
+        }
```

Comment:
is this duplicated with the code snippet in DVCLocalApiClient.java? Can we move this to a common method?

Reply:
good call, i pulled that out into a static util function which is called in the three locations